2996_reflection/reflection.md

```diff
@@ -3642,7 +3643,20 @@ Add a new paragraph prior to the definition of _manifestly constant evaluated_ (
     * in a `$requires-expression$` ([expr.prim.req]{.sref}), or
   * [#.#.#]{.pnum} is, or is a subexpression of, an initializer for a variable that is neither  `constexpr` ([dcl.constexpr]{.sref}) nor `constinit` ([dcl.constinit]{.sref}).
 
-[As detailed below, the evaluation of a plainly constant-evaluated expression may produce new declarations. Their definition precludes expressions that may be evaluated repeatedly.]{.note12}
+[Plainly constant-evaluated expressions are evaluated exactly once, and that evaluation precedes any subsequent parsing ([lex.phases]{.sref}). As detailed below, evaluations of such expressions are allowed to produce injected declarations.]{.note}
+
+::: example
+```cpp
+bool fn();
+consteval bool cfn(int);
+template <int V> requires (cfn(V));  // cfn(V) is not plainly constant-evaluated
+
+constexpr bool b1 = !cfn(1);         // !cfn(1) is plainly constant-evaluated
+const bool b2 = cfn(1);              // cfn(1) is not plainly constant-evaluated
+const bool b3 = cfn(1) && fn();      // cfn(1) is not plainly constant-evaluated
```

Comment:
For clarity, I think better to pass distinct arguments. So `!cfn(1)` for the first one, `cfn(2)` for the second, etc. 

`b3` doesn't add anything on top of `b2` — given that the call isn't plainly constant-evaluated there, the one in `b3` would be... even less so.

I think worth adding a couple more declarations here, including an `if constexpr`? Maybe complete the dangling `requires` up there with something like:
```cpp
consteval bool cfn(int ) { return true; }

template <int V> requires (cfn(V));  // cfn(V) is not plainly constant-evaluated
consteval int g() {
    if constexpr (cfn(V+1)) { // cfn(V+1) is plainly constant-evaluated
        return cfn(V+2); // what about this one?
    } else {
        return 0;
    }
}
```
So for the line marked — `cfn(V+2)` is an immediate invocation that doesn't fit any of those sub-bullets. So that's plainly-constant-evaluated right, even if use a call to like `g<42>()` in an initializer for a variable that is neither `constexpr` nor `constinit` (or itself as part of a `requires` clause)?

Reply:
`cfn(V+2)` is an immediate invocation that results from a substitution of template parameters, _but does **not**_ result from one
* during template argument deduction,
* in a _concept-id_, nor
* in a _requires-expression_

and is therefore plainly constant-evaluated.

Added the suggested examples, thanks!